### DISCUSSION TITLE
New Context Menu Question

### DISCUSSION THREAD

**Discussion by N3C2L** [RyotaUshio/obsidian-pdf-plus#129] [Q&A] [answered]
Thanks a bunch for this new custom context menu! it's very useful!

I love the auto popup of the context menu when the specific setting for this context menu is set to "always".

It's awesome that the colors are now visible in this new context menu.

One small request, or maybe it's a small bug (I don't know therefore I am posting this here in the discussion section):

After highlighting, If I choose another color with another link copy format (another than the default link copy format) from the context menu, then it gets copied successfully to the annotation file.

But when I right click again after highlighting then the previous link copy format is not auto-selected in the previous selected color section again (the default copy format has a check mark again in this color section).

But such a setting would be useful, because then I can only click on the color again after highlighting, and my previous selected link copy format will be copied to the annotation file.

**Comment by RyotaUshio** [comment, marked as answer]
Thanks, I'm glad you liked it!

In fact the behavior is intended, but I agree it's nice have that option and I'm planning to add it shortly.

**Reply by RyotaUshio** [reply]
Will be released in the next version.

**Reply by N3C2L** [reply]
Thank you!

**Reply by RyotaUshio** [reply]
Released as [0.39.6](https://github.com/RyotaUshio/obsidian-pdf-plus/releases/tag/0.39.6).

**Comment by N3C2L** [comment]
Thanks.

I just noticed something.

I enabled the edit to pdf directly option and was highlighting a text.

In the context menu there is now no option to write a highlighted text directly to the PDF without creating a link to the annotation file.

I think such an option is important for people who wants to highlight the text without creating a link in the annotation file.

**Reply by RyotaUshio** [reply]
When I first released the new context menu, I renamed the former "Write highlight to file & copy link" to "Add highlight to file".

Do you mean you don't see the "Add highlight to file" section even though you've enabled PDF editing?
If so, it's strange. I didn't remove it and also I won't do so in the future.

(In my environment, the section still shows up as before:
<img width="1552" alt="image" src="https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/42461ed3-11b8-4209-9a7a-a098ab91b5f1">
)

**Reply by RyotaUshio** [reply]
One possibility is that you might be turning off the "Add highlight to file" toggle switch in the settings:

![Untitled](https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/918e5f86-334c-42d2-a233-1adb0855d94a)

**Comment by N3C2L** [comment]
Oh, sorry. I forgot something to add to my problem.

Yes, I have enabled the setting you described above.

For example, if you right click and choose "quote" (below the "add highlight to file"), then there will be a highlight written to the PDF directly + a link format + a display format will also be written to the annotation .md file.

But what if somebody wants to right click after highlighting and be able to write a highlight to the PDF only but without creating a link format + display format to the annotation file? 

That is not possible now with the actual right click menu but would be a great addition.

**Comment by N3C2L** [comment]
Sorry, my bad. 

This is only the case when auto-paste is activated.

It is working when it is disabled.
